### PR TITLE
Added slideshow functionality

### PR DESCRIPTION
**Pull request by highkite**
Hello,

I added a slideshow feature. By providing a directory instead of a file path with the -i flag it will iteratively show the images that are in the folder. The --slideshow-interval flag can be used to set the time between two images in seconds.

Sorry if the style is not adequate I am not used to program in C, but I am willing to improve it.

All the best,
highkite

### CODE REVIEW

File: i3lock.c
Comment (PandorasFox) [inline i3lock.c]: I'd lump this in with the other misc opts and put it at 903

File: i3lock.c
Comment (PandorasFox) [inline i3lock.c]: ditto as above (903, move it above debug)

File: unlock_indicator.c
Comment (PandorasFox) [inline unlock_indicator.c]: `((lastCheck == -1) || (now - lastcheck >= slideshow_interval))`

realistically I think that `lastCheck` could be set somewhere - maybe in `init_colors_once` or something? I can't think of an ideal place to set it off the top of my head.

File: unlock_indicator.c
Comment (PandorasFox) [inline unlock_indicator.c]: maybe allow for random indexing (i.e. use rand() to generate an index from [0, slideshow_image_count) )

File: i3lock.c
Comment (PandorasFox) [inline i3lock.c]: should also handle jpegs

File: i3lock.c
Comment (PandorasFox) [inline i3lock.c]: this code to load a single image into a buffer will get reused above in the load_slideshow thing (which didn't handle jpegs).

it's probably best to just dump this into a helper function:

```
cairo_surface_t* load_image(char* path) {
// copy code, return `NULL` on fail
}
```

**Review by PandorasFox — COMMENTED** [review]
looks pretty good overall, but could do with a bit of tweaking. It'll be a while before I have enough time to sit down and test these changes.

**Comment by highkite** [comment]
Hi, 

Thank you for your review. I incorporated your comments and at least on my machine it seems to work.
I initialized the `lastCheck` variable in the `init_color_once` to avoid the `-1` check. As a suggestion the method could be renamed to `init_once`, since it comprises now the initialization of multiple components.


All the best,
Highkite

File: i3lock.c
Comment (PandorasFox) [inline i3lock.c]: I think it makes more sense to nest it like:

```
if (image_path != NULL) {
    if (is_regular_file(image_path)) {
        img = load_image(image_path);
    } else {
        load_slideshow_images(image_path);
    }
}
```

Just since there'll need to be a null-check for either of them, really, so we may as well hoist that check out a bit so it makes a bit more sense.

Otherwise, it all looks good and I'll merge it with dev after that.

Don't worry about rebasing on master/dev as they are right now, since I'm pretty used to that by now.

**Review by PandorasFox — COMMENTED** [review]
overall it's solid and I'll merge them + do some more intensive testing on it after this last change - sorry for the delays, I've had some health issues over the past week.

**Comment by highkite** [comment]
Hey,

sorry for the late reply. I was quite busy at work. 
You are absolutely correct with the NULL check. It makes definitely more sens this way. I changed it accordingly.

**Review by PandorasFox — APPROVED** [review]

**Comment by PandorasFox** [comment]
merged it into dev; I'll poke at it a little bit more before merging it into master (I'm trying to be a bit better about that now, oops)

**Pull request merged**
